### ISSUE TITLE
some set methods should accept other type

### ISSUE THREAD

**Issue by jgirardet** [RustPython/RustPython#729]
cpython
```python
>>> a = set([1,2,3,4,5])
>>> a.intersection([1,2]) 
>>> {1, 2}
```

rustpython

```python
>>> a = set([1,2,3,4,5])
>>> a.intersection([1,2]) 
>>> {1, 2}
Traceback (most recent call last):
  File <stdin>, line 0, in <module>
TypeError: Expected type <class 'set'>, not <class 'list'>```

**Comment by jgirardet** [comment]
in cpython other is a pyobject not pysetobject.
I'll look into if if you want

**Comment by coolreader18** [comment]
That'd be great, thanks!

**Comment by palaviv** [comment]
@jgirardet notice I am doing a pretty big change to set in #725

**Comment by jgirardet** [comment]
ok . @palaviv  I see you started adding pyobjectref instead of pysetref in various places. So I sould w& s for now.
btw also changed many "zelf" to "pyobjectref" inspite of it is pysetref in cpython, is that intentional ?

**Comment by palaviv** [comment]
The goal is to use the same code for `set` and `frozenset`. We currently do not support multiple types for the same payload so I had to change back to `pyobjectref`.

**Comment by OddCoincidence** [comment]
There's a `PyIterable` type that can be used to accept anything that conforms to the iterator protocol. I think that's probably what these methods should be using.

**Comment by palaviv** [comment]
This was fixed as part of #785.
